service/index/writer.go

```diff
@@ -334,28 +340,23 @@ func (w *Writer) Close() error {
 func (w *Writer) flush() {
 	defer w.wg.Done()
 
+	ticker := time.NewTicker(w.cfg.FlushInterval)
```

Comment:
I prefer calling `defer ticker.Stop()` as soon as it's declared over calling `ticker.Stop()` manually at the end of the func. This makes the code more robust, should it ever change and include another return path. WDYT?